### PR TITLE
mgr/cephadm: fix osd reconfig/redeploy

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -1253,12 +1252,8 @@ def _create_osd(self, all_hosts_, drive_group):
                     continue
 
                 # create
-                ret, keyring, err = self.mon_command({
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py]: Just to clarify. Moving this doesn't fix any bugs, right?

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -1250,13 +1258,19 @@ def _create_osd(self, all_hosts_, drive_group):
                 if len(list(set(devices) & set(osd['devices']))) == 0 and osd.get('lv_path') not in devices:
                     self.log.debug('mismatched devices, skipping %s' % osd)
                     continue
+                if osd_id not in osd_uuid_map:
+                    self.log.debug('osd id %d does not exist in cluster' % osd_id)
+                    continue
+                if osd_uuid_map[str(osd_id)] != osd['tags']['ceph.osd_fsid']:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py]: `osd_uuid_map` is of type `Dict[str, str]`? 

And according to 

https://github.com/ceph/ceph/blob/afc28b96de738dbca674e5d7c34660f1de2b1ab4/src/ceph-volume/ceph_volume/devices/lvm/listing.py#L243

`osd_id` is already a `str`. can we simply drop `str()` here? Right now this looks as if `osd_id` is an integer.

File: src/pybind/mgr/cephadm/module.py [hunk 3]
```diff
@@ -1320,6 +1337,15 @@ def _create_daemon(self, daemon_type, daemon_id, host,
                      'mgr', 'profile crash'],
         })
 
+        # osd deployments needs an --osd-uuid arg
+        if daemon_type == 'osd':
+            if not osd_uuid_map:
+                osd_uuid_map = self.get_osd_uuid_map()
+            osd_uuid = osd_uuid_map.get(daemon_id)
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py]: This raises a ValueErorr, if `daemon_id not in osd_uuid_map`, which makes the code below unreachable. 

```suggestion
            osd_uuid = osd_uuid_map.get(daemon_id, None)
```

In general, can we move this code block upwards to the redeploy case?

File: src/pybind/mgr/cephadm/module.py (hunk 1 above)
Comment (liewegas) [reply src/pybind/mgr/cephadm/module.py]: right, just refactoring a bit

**Pull request merged**